obspy/core/tests/test_stream.py

```diff
@@ -2656,6 +2656,59 @@ def test_write_empty_stream(self):
             self.assertEqual(e.exception.args[0],
                              'Can not write empty stream to file.')
 
+    def test_stack(self):
+        """
+        Tests stack method
+        """
+        st = read()
+        st2 = st.stack()
+        self.assertEqual(len(st2), 1)
+        self.assertEqual(st2[0].stats.stack, 'all')
+        self.assertEqual(st2[0].stats.stack_count, 3)
+        st += st.copy()
+        st2 = st.stack('seedid')
+        self.assertEqual(len(st2), 3)
+        self.assertEqual({tr.stats.stack for tr in st2}, {tr.id for tr in st})
```

Comment:
Should this preserve order? If so use a list or tuple rather than set. Same for other set asserts below.

Reply:
No, the order is not important. It will only be preserved on python>3.4. Also, the number of traces of the stack and the original stream is different, so list or tuple would not be sufficient. The keys of an OrderedDict would be.